node/core/backing/src/lib.rs

```diff
@@ -1133,21 +1135,44 @@ async fn seconding_sanity_check<Context>(
 	}
 
 	let mut membership = Vec::new();
-	let mut responses = FuturesOrdered::new();
+	let mut responses = FuturesOrdered::<BoxFuture<'_, Result<_, oneshot::Canceled>>>::new();
```

Comment:
nit: use `Either` to avoid allocation and virtualization.

Reply:
This will cause issues with deriving lifetimes: output of both `Either` variants should have the same type, but we implicitly borrow head and leaf state